tutorials/platform/html5_shell_classref.rst

```diff
@@ -0,0 +1,266 @@
+
+    :param number majorVersion:
+        The major WebGL version to check for. Defaults to ``1`` for *WebGL 1.0*.
+    :returns:
+        ``true`` if the given major version of WebGL is available, ``false``
+        otherwise.
+
+.. js:method:: Engine.setWebAssemblyFilenameExtension(extension)
+
+    Set an alternative filename extension for WebAssembly module. By default 
+    it is assumed to be ``wasm``.
+
+    :param string extension:
+        Filename extension without preceding dot.
+
+
+Instance Property Descriptions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. js:attribute:: engine.rtenv
+
+    The runtime environment provided by Emscripten's ``Module``. For more information
+    refer to the `official documentation <https://emscripten.org/docs/api_reference/module.html>`__ on Emscripten.
+    
+Instance Method Descriptions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. js:class:: Engine
+
+    Create a new instance of ``Engine`` class
+
+.. js:method:: engine.init([basePath])
+
+    Initialize the engine instance. Optionally, pass the base path to the engine to load it,
+    if it hasn't been loaded yet. See :js:meth:`Engine.load`.
+
+    :param string basePath:
+        Base path of the engine to load.
+
+    :returns:
+        Promise that resolves once the engine is loaded and initialized.
+
+.. js:method:: engine.preloadFile(file[, path])
+
+    Load a file so it is available in the instance's file system once it runs. Must
+    be called **before** starting the instance.
+
+    :param string|ArrayBuffer file:
+        If type is ``string``, the file will be loaded from that path.
+
+        If type is ``ArrayBuffer`` or a view on one, the buffer will used as
+        the content of the file.
+
+    :param string path:
+        Path by which the file will be accessible. Required, if ``file`` is not
+        a string. If not passed, the path is derived from the URL of the loaded
+        file.
+
+    :returns:
+        Promise that resolves once the file is loaded.
+
+.. js:method:: engine.start([arg1, arg2, …])
+
+    Start the instance of the engine, using the passed strings as
+    command line arguments. :js:meth:`engine.startGame` can be used 
+    in typical cases instead.
+    
+    This will initialize the instance, if it is not initialized. For manual 
+    initialization see :js:meth:`engine.init`. The engine must be loaded beforehand.
+
+    Fails, if a canvas cannot be found on the page.
+
+    :param string variadic:
+        Command line argument.
+
+    :returns:
+        Promise that resolves once the engine started.
+
+.. js:method:: engine.startGame(execName, mainPack)
+
+    Start the game instance using given executable URL and main pack URL.
+
+    This will initialize the instance, if it is not initialized. For manual 
+    initialization see :js:meth:`engine.init`.
+
+    This will load the engine, if it is not loaded. The base path of the 
+    executable URL will be used as the engine base path.
```

Comment:
```suggestion
    This will load the engine if it is not loaded. The base path of the 
    executable URL will be used as the engine base path.
```